### PR TITLE
Add ProviderID to KubemarkMachine

### LINKED ISSUE #53: "providerID" should be on the kubemarkMachine object
When creating nodes with the Cluster API Kubemark I see errors on the CAPI reconciler because the provider ID is not set on the KubemarkMachine i.e. this error: https://github.com/kubernetes-sigs/cluster-api/blob/dba69fff871b8d2f0c3ade87043729f97916dbb5/internal/controllers/machine/machine_controller_phases.go#L296

`spec.providerID` doesn't exist in the KubemarkMachineSpec, so it should be added and then set during reconcile (my assumption, correct me if I'm wrong, is that a dummy value will work for this field).

`spec.providerID` should be set on all infrastructureMachine types according to: https://cluster-api.sigs.k8s.io/developer/providers/machine-infrastructure.html

I'm happy to implement this, but I'm wondering if I'm missing something that explains why this hasn't been implemented previously.

/kind bug

### PR DESCRIPTION
**Pull request by killianmuldoon**
Set ProviderID on the KubemarkMachine during reconcile. 

Fixes #53

### CODE REVIEW

File: controllers/kubemarkmachine_controller.go [hunk 1]
```diff
@@ -409,7 +409,8 @@ func (r *KubemarkMachineReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		}
 	}
 
-	machine.Spec.ProviderID = pointer.StringPtr(fmt.Sprintf("kubemark://%s", kubemarkMachine.Name))
+	machine.Spec.ProviderID = pointer.String(fmt.Sprintf("kubemark://%s", kubemarkMachine.Name))
+	kubemarkMachine.Spec.ProviderID = pointer.String(fmt.Sprintf("kubemark://%s", kubemarkMachine.Name))
```
Comment (elmiko) [inline controllers/kubemarkmachine_controller.go:413]: any reason not to do the `fmt.Sprintf` once and assign to a var?

File: api/v1alpha4/kubemarkmachine_types.go [hunk 2]
```diff
@@ -38,6 +38,10 @@ const (
 
 // KubemarkMachineSpec defines the desired state of KubemarkMachine
 type KubemarkMachineSpec struct {
+	// ProviderID will be the kubemark pod in ProviderID format (kubemark:////<podname>)
```
Comment (elmiko) [inline api/v1alpha4/kubemarkmachine_types.go:41]: minor nit
```suggestion
	// ProviderID will be the kubemark pod name in ProviderID format (kubemark:////<podname>)
```

**Review by elmiko — COMMENTED** [review]
looks mostly good to me, just a minor nit and a question

File: controllers/kubemarkmachine_controller.go (hunk 1 above)
Comment (killianmuldoon) [reply controllers/kubemarkmachine_controller.go:413]: Good idea - updated.

**Review by elmiko — COMMENTED** [review]
thanks @killianmuldoon 

/lgtm
/approve

**Pull request merged**